Issue #88 Disable filtering when destroying

gjanssens opened the pull request:
The code internally could not distinguish between setting -P or not setting any -p
option at all. The net result was that -P really only unsets all prefixes passed
on the command line with -p 'prefix' before the -P is encountered.

This commit changes that: -P now is treated independently and will skip
prefix filtering if set.

graudeejs commented:
Tests are broken with this change :(

The problem is that `PREFIX_FILTER='true'` should be defined in core since you modify core logic.

gjanssens commented:
Thanks for the feedback. I will have a look.

gjanssens commented:
I have slightly rewritten the logic. First I have reverted the logic - there's no SKIP_PREFIX_FILTER instead of PREFIX_FILTER. This made the conditionals slightly easier to construct. Next I have added a local default in core. So if SKIP_PREFIX_FILTER is unset, core acts as if it's 'false'. And finally I have added unit tests to verify it all still works.
